src/Widgets/ColumnContainerWidget/ColumnContainerWidgetComponent.tsx

```diff
@@ -1,33 +1,76 @@
-import { provideComponent, ContentTag } from 'scrivito'
+import { provideComponent, ContentTag, connect } from 'scrivito'
 import { ColumnContainerWidget } from './ColumnContainerWidgetClass'
+import { ColumnWidgetInstance } from '../ColumnWidget/ColumnWidgetClass'
 
 provideComponent(ColumnContainerWidget, ({ widget }) => {
   const columns = widget.get('columns')
   if (!columns.length) return null
 
-  const content = columns.map((columnWidget, index) => {
-    const colSize = columnWidget.get('colSize') || 1
-    const className = widget.get('disableResponsiveAdaption')
-      ? `col-${colSize}`
-      : `col-md-${colSize}`
-    return (
-      <div key={index} className={className}>
-        <ContentTag
-          content={columnWidget}
-          attribute="content"
-          className="h-100"
-        />
-      </div>
-    )
-  })
-
-  const classNames = ['row']
-
-  if (widget.get('alignment')) {
-    classNames.push(`align-items-${widget.get('alignment')}`)
-  } else {
-    classNames.push('align-items-start')
+  const alignment = widget.get('alignment') || 'start'
+  const isResponsive = !widget.get('disableResponsiveAdaption')
+  const isFlex = widget.get('layoutMode') === 'flex'
+
+  const classNames = [`align-items-${alignment}`]
+  const flexClassName = isResponsive ? 'd-md-flex' : 'd-flex'
+  classNames.push(isFlex ? flexClassName : 'row')
+
+  return (
+    <div className={classNames.join(' ')}>
+      {columns.map((columnWidget: ColumnWidgetInstance) => {
+        const Column = isFlex ? FlexColumn : GridColumn
+        return (
+          <Column
+            key={columnWidget.id()}
+            columnWidget={columnWidget}
+            isResponsive={isResponsive}
+            isStretch={alignment === 'stretch'}
+          />
+        )
+      })}
+    </div>
+  )
+})
+
+type ColumnProps = {
+  columnWidget: ColumnWidgetInstance
+  isResponsive: boolean
+  isStretch: boolean
+}
+
+const FlexColumn = connect(function FlexColumn({
+  columnWidget,
+  isResponsive,
+  isStretch,
+}: ColumnProps) {
+  const classNames = isResponsive ? ['my-md-0', 'm-2'] : ['mx-2']
+  if (columnWidget.get('flexGrow')) classNames.push('flex-grow-1')
+  if (isStretch && columnWidget.get('content').length < 2) {
+    classNames.push(isResponsive ? 'd-md-flex' : 'd-flex')
```

Comment:
* [ ] This seems to not cover the full height (see middle column).
* [ ] Also this seems to make the left and middle column have less space available - which is unexpected to me:

<img width="1624" alt="Screenshot 2024-08-06 at 09 31 34" src="https://github.com/user-attachments/assets/a9d6c3da-7135-42fe-a888-a11d1e2e27d1">
<img width="1624" alt="Screenshot 2024-08-06 at 09 31 30" src="https://github.com/user-attachments/assets/cfc14fa4-ebc8-4722-9d1b-88ddae94ce19">

Reply:
> * [ ]  This seems to not cover the full height (see middle column).

* Our card widget has a bottom margin
* For grid display we use a workaround of a `h-100` wrapper, which through some CSS quirk moves the margin of the card to an invisible location, but only on desktop. In the **responsive** view the issue is **also visible for grid**.
* For flex the workaround will not work, b/c flex has no support for `h-100`.

So, actually I'd say the (only) unexpected behavior are desktop grid columns.
Maybe we should rethink card (and other?) widget margins in general and make them work in all cases, grid, flex, mobile, desktop.

> * [ ]  Also this seems to make the left and middle column have less space available - which is unexpected to me:

This is how flex works and how it's supposed to be for now. But I see that an average editor may expect a different
outcome. We should loop this feedback to Alex. Maybe there is some flex magic that leads to a more deterministic layout.